### DISCUSSION TITLE
How to pass `csl` option to `pandoc.utils.citeproc`

### DISCUSSION THREAD

**Discussion by urosmikanovic** [jgm/pandoc#9578] [Q&A] [answered]
I'm trying to process some citations in a lua filter and my `csl` option doesn't seem to have an effect on `pandoc.utils.citeproc`. It renders citations in the (I imagine) default style. I would also expect it to be a key in `PANDOC_WRITER_OPTIONS` but it doesn't seem to be.

Does anyone have an example filter that does this? Thanks!

**Comment by jgm** [comment, marked as answer]
The csl option just adds a `csl` field to the document's metadata.  Can you confirm from inside your filter that your metadata contains this field?

**Reply by urosmikanovic** [reply]
Thanks! Can confirm and `pandoc.utils.citeproc` uses the specified csl file if it's in the document's Meta field.

I see it's and XY problem now. When opening, reading, and writing other sources inside my lua filter the option is missing in these additional documents' metadata.

I thought it was a writer option, or didn't think it was a metadata field because in the defaults file it's not set as a key under metadata.

```yaml
---
citeproc: true
csl: foo.csl
metadata:
  key: val
...
```

**Reply by jgm** [reply]
All you need to do, then, is add this to the document's metadata before calling citeproc.
